### PR TITLE
build(client): Tag asserts

### PR DESCRIPTION
**Pull request by tylerbutler**
Command used:

```shell
pnpm policy-check:asserts
```

### CODE REVIEW

**Comment by anthony-murphy** [comment]
the failing tests here look like improper usages of asserts. @tyler-cai-microsoft  and @jatgarg. Asserts should never be hit as they represent failed invariants, specifically if they are hit it is a bug. it looks like these test are validating user input with asserts. this is the wrong usage of assert. you should update the code to throw user errors

**Review by markfields — APPROVED** [review]

File: packages/loader/container-loader/src/test/serializedStateManager.spec.ts [hunk 1]
```diff
@@ -176,30 +176,6 @@ describe("serializedStateManager", () => {
 		);
 	});
 
-	it("can't get pending local state when there is no base snapshot", async () => {
```
Comment (ChumpChief) [inline packages/loader/container-loader/src/test/serializedStateManager.spec.ts:179]: This deletion is intentional?  From a procedural perspective, might be nice to split any material changes into their own PR to go before the tagging.

**Review by ChumpChief — APPROVED** [review]

File: packages/loader/container-loader/src/test/serializedStateManager.spec.ts (hunk 1 above)
Comment (tylerbutler) [reply packages/loader/container-loader/src/test/serializedStateManager.spec.ts:179]: Daniel's got a PR for this change: https://github.com/microsoft/FluidFramework/pull/20054

I added it here to make sure this PR passes with that fix. His change will get merged first.

**Review by CraigMacomber — APPROVED** [review]
Tree changes look fine.

**Review by scarlettjlee — APPROVED** [review]
